visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -370,7 +390,10 @@ public void sauceVisualCheck(String snapshotName, CheckOptions options) {
       input.setClipSelector(clipSelector);
     }
 
-    input.setFullPageConfig(options.getFullPageScreenshotConfig());
+    FullPageScreenshotConfig fullPageScreenshotConfig = Optional.ofNullable(options.getFullPageScreenshotConfig()).orElse(this.fullPageScreenshotConfig);
+    if (fullPageScreenshotConfig != null) {
+      input.setFullPageConfig(fullPageScreenshotConfig);
+    }
```

Comment:
Wouldn't something like this be simplier ?
```suggestion
    input.setFullPageConfig(
      Optional.ofNullable(options.getFullPageScreenshotConfig()).orElse(this.fullPageScreenshotConfig)
    );
```

Reply:
I think the intention is not to set it on `input` if `this.fullPageScreenshotConfig` is null. It doesn't read great, but it follows the way it was done `captureDom`